src/lambda/vue/samInvoke.ts

```diff
@@ -450,3 +426,47 @@ function getLaunchConfigQuickPickItems(launchConfig: LaunchConfiguration, uri: v
             }
         })
 }
+
+function pruneConfig(config: AwsSamDebuggerConfiguration, name: string): AwsSamDebuggerConfiguration {
+    const newConfig = pruneConfigHelper(config)
+    newConfig.name = name
+
+    if (isTemplateTargetProperties(config.invokeTarget)) {
+        newConfig.invokeTarget = {
+            target: config.invokeTarget.target,
+            logicalId: config.invokeTarget.logicalId,
+            templatePath: config.invokeTarget.templatePath,
+        }
+    } else if (isCodeTargetProperties(config.invokeTarget)) {
+        newConfig.invokeTarget = {
+            target: config.invokeTarget.target,
+            lambdaHandler: config.invokeTarget.lambdaHandler,
+            projectRoot: config.invokeTarget.projectRoot,
+        }
+    }
+
+    return newConfig
+}
+
+function pruneConfigHelper(object: { [key: string]: any }): any | undefined {
+    const keys = Object.keys(object)
+    const final: any = {}
+    for (const key of keys) {
+        const val = object[key]
+        if (val !== undefined && val !== '' && !(Array.isArray(val) && val.length === 0)) {
+            if (typeof val === 'object') {
+                const pruned = pruneConfigHelper(val)
+                if (pruned) {
+                    final[key] = pruned
```

Comment:
IIUC, this function works by building a new object and only including non-empty fields.

It might be less verbose to use this approach instead:

1. create a deep copy via `const final = JSON.parse(JSON.stringify(...))`
2. remove the empty fields via `delete`: https://stackoverflow.com/a/38364486/152142

Reply:
Was thinking about delete but didn't want to accidentally delete something tied to the main object. `const final = JSON.parse(JSON.stringify(...))` could work though, I'll give it a shot.

Unsure what circular references could arise from this, though? JSON shouldn't be able to refer to things not nested within it, no?

Reply:
> Unsure what circular references could arise from this, though? JSON shouldn't be able to refer to things not nested within it, no?

Yep, as long as nothing in the codepath changed the object. Minor concern here.